pandas/core/strings.py

```diff
@@ -1314,21 +1314,65 @@ def str_index(arr, sub, start=0, end=None, side='left'):
 
 def str_pad(arr, width, side='left', fillchar=' '):
     """
-    Pad strings in the Series/Index with an additional character to
-    specified side.
+    Pad strings in the Series/Index up to width.
 
     Parameters
     ----------
     width : int
         Minimum width of resulting string; additional characters will be filled
-        with spaces
+        with character defined in fillchar.
     side : {'left', 'right', 'both'}, default 'left'
-    fillchar : str
-        Additional character for filling, default is whitespace
+        Side from which to fill resulting string.
+    fillchar : str, default ' '
+        Additional character for filling, default is whitespace.
 
     Returns
     -------
-    padded : Series/Index of objects
+    Series or Index of object
+        Returns Series or Index with minimum number of char in object.
+
+    Examples
+    --------
+    >>> s = pd.Series(["caribou", "tiger"])
+    >>> s
+    0    caribou
+    1      tiger
+    dtype: object
+
+    >>> s.str.pad(width=10)
+    0       caribou
+    1         tiger
+    dtype: object
+
+    >>> s.str.pad(width=10, side='right', fillchar='-')
+    0    caribou---
+    1    tiger-----
+    dtype: object
+
+    >>> s.str.pad(width=10, side='both', fillchar='-')
+    0    -caribou--
+    1    --tiger---
+    dtype: object
+
+    >>> s = pd.Series(["caribou", 0, np.NaN])
+    >>> s.str.pad(width=10, fillchar='-')
+    0    ---caribou
+    1           NaN
+    2           NaN
+    dtype: object
```

Comment:
Not sure if I'd include the `0` and `np.nan` in the original Series, so we don't need this last example.

We usually use `np.nan` instead of `np.NaN`, I'd change it for consistency.